src/utils/get-changed-fields.ts

```diff
@@ -0,0 +1,26 @@
+type Nullable<T> = {
+  [K in keyof T]: T[K] | null
+}
+
+export function getChangedFields<T>(
```

Comment:
Why you didn't specify a more detailed generic type here? Just `T` can be anything, I can pass here a plain string, array, or something else. I think we should make that forbidden here.

```suggestion
export function getChangedFields<T extends Record<string, unknown>>(
```